versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTablePersist.java

```diff
@@ -540,6 +531,81 @@ public void upsertObjs(@Nonnull Obj[] objs) throws ObjTooLargeException {
     }
   }
 
+  @Override
+  public boolean deleteConditional(@Nonnull UpdateableObj obj) {
+    ConditionalRowMutation conditionalRowMutation =
+        mutationForConditional(obj, Mutation.create().deleteRow());
+    return backend.client().checkAndMutateRow(conditionalRowMutation);
+  }
+
+  @Override
+  public boolean updateConditional(@Nonnull UpdateableObj expected, @Nonnull UpdateableObj newValue)
+      throws ObjTooLargeException {
+    ObjId id = expected.id();
+    checkArgument(id != null && id.equals(newValue.id()));
+    checkArgument(expected.type().equals(newValue.type()));
+    checkArgument(!expected.versionToken().equals(newValue.versionToken()));
+
+    Mutation mutation = objectWriteMutation(newValue, false);
+
+    ConditionalRowMutation conditionalRowMutation = mutationForConditional(expected, mutation);
+    return backend.client().checkAndMutateRow(conditionalRowMutation);
+  }
+
+  @NotNull
+  private ConditionalRowMutation mutationForConditional(
+      @NotNull UpdateableObj obj, Mutation mutation) {
+    checkArgument(obj.id() != null, "Obj to store must have a non-null ID");
+    ByteString key = dbKey(obj.id());
+
+    Filter condition =
+        FILTERS
+            .chain()
+            .filter(
+                FILTERS
+                    .chain()
+                    .filter(FILTERS.qualifier().exactMatch(QUALIFIER_OBJ_VERS))
+                    .filter(FILTERS.value().exactMatch(obj.versionToken())))
+        // TODO adding this safety net doesn't work :(
```

Comment:
Try this:

```java
ChainFilter objTypeFilter =
  FILTERS
      .chain()
      .filter(FILTERS.qualifier().exactMatch(QUALIFIER_OBJ_TYPE))
      .filter(FILTERS.value().exactMatch(OBJ_TYPE_VALUES.get(obj.type())));
ChainFilter objVersionFilter =
  FILTERS
    .chain()
    .filter(FILTERS.qualifier().exactMatch(QUALIFIER_OBJ_VERS))
    .filter(FILTERS.value().exactMatch(obj.versionToken()));
Filter condition =
    FILTERS.condition(objTypeFilter).then(objVersionFilter).otherwise(FILTERS.block());
```

Reply:
ah! this works! thanks!